packages/matchbox/src/components/ListBox/ListBox.js

```diff
@@ -0,0 +1,212 @@
+    label,
+    labelHidden,
+    error,
+    errorInLabel,
+    required,
+    optional,
+    helpText,
+    defaultValue,
+    value,
+    onChange,
+    ...rest
+  } = props;
+
+  const systemProps = pick(rest, system.propNames);
+  const componentProps = omit(rest);
+
+  const [open, setOpen] = useState(false);
+  const [currentValue, setCurrentValue] = React.useState(
+    value ? value : defaultValue != null ? defaultValue : placeholder,
+  );
+
+  const { describedBy, errorId, helpTextId } = useInputDescribedBy({
+    id,
+    hasHelpText: !!helpText,
+    hasError: !!error,
+  });
+
+  const requiredIndicator = required ? (
+    <Box as="span" pr="200" aria-hidden="true">
+      *
+    </Box>
+  ) : null;
+
+  function togglePopover(event) {
+    event.stopPropagation();
+    setOpen(!open);
+  }
+
+  function handleClose() {
+    setOpen(false);
+  }
+
+  function onSelect(value) {
+    if (onChange) {
+      onChange(value);
+    }
+
+    setOpen(false);
+    setCurrentValue(value);
+  }
+
+  const contentFromValue = React.useMemo(() => {
+    let options = getChild('ListBox.Option', children);
+    let activeOption = options.find(option => {
+      return option.props.value === currentValue;
+    });
+
+    return activeOption ? activeOption.props.children : currentValue;
+  }, [currentValue]);
+
+  const labelMarkup = (
+    <Label id={id} label={label} labelHidden={labelHidden}>
+      {requiredIndicator}
+      {error && errorInLabel && (
+        <Box as={Error} id={errorId} wrapper="span" error={error} fontWeight="400" />
+      )}
+      {optional && <OptionalLabel float />}
+    </Label>
+  );
+
+  const helpMarkup = helpText ? <HelpText id={helpTextId}>{helpText}</HelpText> : null;
+
+  let options = React.useMemo(() => {
+    return getChild('ListBox.Option', children);
+  }, [children]);
+
+  const { optionsMarkup, focusContainerProps } = useOptionConstructor({
+    options,
+    value: currentValue,
+    onSelect,
+    open,
+    placeholder,
+  });
+
+  return (
+    <StyledWrapper tabIndex="-1" {...systemProps} {...focusContainerProps}>
+      {labelMarkup}
+      <Popover
+        id="listbox-popover"
+        p="0"
+        onClose={handleClose}
+        open={open}
+        as="div"
+        width="100%"
+        trigger={
+          <Box position="relative">
+            <StyledButton
+              id={id}
+              data-id="open-listbox"
+              textAlign="left"
+              fullWidth
+              variant="mutedOutline"
+              aria-haspopup="listbox"
+              aria-pressed={open}
+              aria-expanded={open}
+              aria-labelledby={id}
+              onClick={togglePopover}
+              disabled={disabled}
+              {...componentProps}
+              {...describedBy}
+            >
+              {contentFromValue}
+            </StyledButton>
+            <StyledChevron size={24} disabled={disabled} />
+          </Box>
+        }
+      >
+        <StyledList
```

Comment:
I think we need to add a max height somehow

![Screen Shot 2020-09-10 at 10 03 23 AM](https://user-images.githubusercontent.com/3903325/92741721-e566b900-f34c-11ea-9686-5cb3a84b3883.png)